setup.py

```diff
@@ -97,6 +103,9 @@
     'License :: OSI Approved :: Apache Software License',
 ]
 
+BUILD_WITH_BORING_SSL_ASM = os.environ.get('GRPC_BUILD_WITH_BORING_SSL_ASM',
```

Comment:
Note that `XYZ = os.environ.get('GRPC_BUILD_WITH_BORING_SSL_ASM', True)` followed by `if XYZ:` test is fundamentally broken. I just ran into this.

even if you assign the string 'False', 'false', '0' or something like that, the non-empty string still evaluates as 'True'. In combination with the 'True' default value, there is now basically now way to disable this option.
(For options with default value `False` you can at least leave the env variable empty  and that will evaluate as false).

Reply:
sorry about that.  I think this was an oversight because I originally had this as false.  Do you need me to patch this?

Reply:
np, but yeah a patch would be nice. Btw the way all the  options are read from env variables is also kind of broken, but they at least can be used because they have default value `False`.


Ideally we'd replace the reading of boolean values from env variables everywhere (if you assign 'XYZ=False' or 'XYZ=false' that's still evaluated as `True` :-( )
https://github.com/grpc/grpc/blob/711d1d7fe2da56896957e00fea12c84b6d1fa1e1/setup.py#L109-L169

@lidizheng  since you reviewed this.

Reply:
Looks like there's already an issue for this: https://github.com/grpc/grpc/issues/24498